src/puppetlabs/trapperkeeper/services/webserver/jetty9_core.clj

```diff
@@ -270,25 +273,34 @@
   (connector* server config nil))
 
 (schema/defn ^:always-validate
-  queue-thread-pool :- QueuedThreadPool
-  [max-threads :- schema/Int
-   queue-max-size :- schema/Int]
-  (let [queue-min-size (min config/default-queue-min-threads queue-max-size)]
-    (QueuedThreadPool. max-threads
-                       queue-min-size
-                       config/default-queue-idle-timeout
-                       (BlockingArrayQueue.
-                         queue-min-size
-                         queue-min-size
-                         queue-max-size))))
+  queue-thread-pool :- (schema/maybe QueuedThreadPool)
+  [max-threads :- (schema/maybe schema/Int)
+   queue-max-size :- (schema/maybe schema/Int)]
+  (if (or max-threads queue-max-size)
+    (let [thread-pool (if max-threads
+                        (QueuedThreadPool. max-threads)
+                        (QueuedThreadPool.))]
+      (if queue-max-size
+        (let [min-threads    (.getMinThreads thread-pool)
```

Comment:
Hmm.. so this is just for 'getMaxThreads' and 'getIdleTimeout'?  Do either of those involve any fancy computations or are they just hard-coded numeric values in Jetty?

If the latter, I'm thinking maybe it would be a lesser evil to just put those hard coded values up into our defaults section with some comments explaining why they're there, and a reference to the tests that validate them... and then a comment next to those tests explaining that if they fail in the future then the user needs to update the defaults?  Or maybe the tests should just be written to use those hard-coded vars in their implementation?

I know that goes against the grain of what we've been trying to do here, but it seems maybe slightly less bad than having to construct a thread pool just to throw it away?